sys/linux/filesystem.txt

```diff
@@ -192,8 +189,24 @@ uuid_str {
 
 hex_chars = '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'
 
-ima_funcs = "FILE_CHECK", "PATH_CHECK", "MODULE_CHECK", "FIRMWARE_CHECK", "FILE_MMAP", "MMAP_CHECK", "BPRM_CHECK", "CREDS_CHECK", "KEXEC_KERNEL_CHECK", "KEXEC_INITRAMFS_CHECK", "POLICY_CHECK"
+ima_action = "measure", "dont_measure", "appraise", "dont_appraise", "audit", "hash", "dont_hash"
+ima_funcs = "FILE_CHECK", "PATH_CHECK", "MODULE_CHECK", "FIRMWARE_CHECK", "FILE_MMAP", "MMAP_CHECK", "BPRM_CHECK", "CREDS_CHECK", "KEXEC_KERNEL_CHECK", "KEXEC_INITRAMFS_CHECK", "POLICY_CHECK", "KEXEC_CMDLINE", "KEY_CHECK"
 ima_masks = "MAY_EXEC", "MAY_WRITE", "MAY_READ", "MAY_APPEND", "^MAY_EXEC", "^MAY_WRITE", "^MAY_READ", "^MAY_APPEND"
+ima_appraise_type = "imasig", "imasig|modsig"
+ima_template = "ima", "ima-ng", "ima-sig", "ima-buf", "ima-modsig"
+# from security/integrity/digsig.c keyring_name
+# TODO: other values? .persistent_register, .dns_resolver?
```

Comment:
@ebiggers do you know anything about this by any chance?

Reply:
Not very much.  If this API is accepting a keyring name (a.k.a. a "key description"), it might be good to add `key_desc` from `sys/linux/key.txt`.

Reply:
key_desc contains '\0' in the end. I can't use it unmodified in policy because of this.
```
key_desc {
prefix  stringnoz["syz"]
id      proc[' ', 4, int8]
z       const[0, int8]
}
```
I can split the structure to key_desc and key_descz to be able to use it in security_ima.txt

Reply:
Splitting key_desc looks reasonable.